support-frontend/test/admin/settings/SwitchesSpec.scala

```diff
@@ -0,0 +1,150 @@
+          |        "state" : "On"
+          |      },
+          |      "sepa" : {
+          |        "description" : "SEPA",
+          |        "state" : "Off"
+          |      }
+          |    }
+          |  },
+          |  "subscriptionsSwitches": {
+          |    "description": "Subscriptions",
+          |    "switches": {
+          |      "checkoutPostcodeLookup" : {
+          |        "description" : "Enable external service postcode lookup in checkout form",
+          |        "state" : "On"
+          |      },
+          |      "enableDigitalSubGifting" : {
+          |        "description" : "Enable Digital Sub gifting",
+          |        "state" : "On"
+          |      },
+          |      "useDotcomContactPage" : {
+          |        "description" : "Use Dotcom contact page",
+          |        "state" : "On"
+          |      }
+          |    }
+          |  },
+          |  "featureSwitches" : {
+          |    "description" : "Feature switches",
+          |    "switches" : {
+          |      "enableQuantumMetric" : {
+          |        "description" : "Enable quantum metric",
+          |        "state" : "On"
+          |      },
+          |      "usStripeAccountForSingle" : {
+          |        "description" : "US Stripe account for single contributions",
+          |        "state" : "On"
+          |      }
+          |    }
+          |  },
+          |  "campaignSwitches" : {
+          |    "description" : "Campaign switches",
+          |    "switches" : {
+          |      "enableContributionsCampaign" : {
+          |        "description" : "Enable contributions campaign",
+          |        "state" : "Off"
+          |      },
+          |      "forceContributionsCampaign" : {
+          |        "description" : "Force all users into the contributions campaign (ignore url path)",
+          |        "state" : "Off"
+          |      }
+          |    }
+          |  },
+          |  "recaptchaSwitches": {
+          |    "description" : "Recaptcha switches",
+          |    "switches" : {
+          |      "enableRecaptchaBackend" : {
+          |        "description" : "Enable Recaptcha on the backend",
+          |        "state" : "On"
+          |      },
+          |      "enableRecaptchaFrontend" : {
+          |        "description" : "Enable Recaptcha on the client",
+          |        "state" : "On"
+          |      }
+          |    }
+          |  }
+          |}
+          |""".stripMargin
+
+      decode[Switches](json) mustBe (Right(
+        Switches(
+          oneOffPaymentMethods = OneOffPaymentMethodSwitches(),
+          recurringPaymentMethods = RecurringPaymentMethodSwitches(),
+          subscriptionsSwitches = SubscriptionsSwitches(),
+          featureSwitches = FeatureSwitches(On, On),
+          campaignSwitches = CampaignSwitches(),
+          recaptchaSwitches = RecaptchaSwitches()
+        )
+      ))
+    }
+  }
+}
```

Comment:
is it worth having an IT-test to read the actual switches file also?

Reply:
With this implementation, these are the possible scenarios:
1. no json file in S3 - server does not come up.
2. malformed json, or a missing switchgroup in the json - server does not come up.
3. a switch is missing - server comes up and uses the default for that switch.

As I write this, I'm doubting the usefulness of the 3rd point here. If a switch is missing, should it instead fail to come up?

Reply:
I feel like fail to come up is slightly better for the 3rd point - I remember the time someone accidentally deployed a really old version of frontend, which used default values for a lot of switches and broke the website quite badly.

In our case we could deploy and thought we have provided a switch but it has a typo in the name, and whoever switches it on in the US or wherever would find the switch had no effect.  So not terrible, but not ideal.

The other side is if the switches file is broken in some way we wouldn't be able to deploy or replace unhealthy instances until it was fixed.

Reply:
👍 I'll remove the defaults and treat missing switches as an error